doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_suit_recovery.rst

```diff
@@ -236,3 +236,177 @@ To turn an application into a recovery application, the following steps have to
     The value of ``SUIT_RECOVERY_APPLICATION_PATH`` can contain variables like ``${ZEPHYR_NRF_MODULE_DIR}``
 
 #. When building the main application, set ``SB_CONFIG_SUIT_RECOVERY_APPLICATION_CUSTOM`` (or the Kconfig option name if a different one was chosen) to ``y``.
+
+.. _ug_nrf54h20_suit_recovery_enter_recovery_request:
+
+Entering recovery via are a request from the local domains
+**********************************************************
+
+It is possible to enter recovery mode via a request sent from the local domains.
+This can be used for several purposes, such as:
+
+* Recovering the device if the main firmware crashes and no JLink connection is available.
+* Using the recovery application as a "foreground update" application, allowing to save some space in the main application.
+* Debugging the recovery application.
+
+:ref:`Entering recovery via button press <ug_nrf54h20_suit_recovery_enter_via_button>` uses such requests.
+The code for this feature is located in the :file:`nrf/subsys/suit/app_tools/recovery_button` directory.
+It can be used as a reference design.
+
+To request entering the recovery mode, the ``suit_foreground_dfu_required()`` function must be called.
```

Comment:
```suggestion
For additional details, see :ref:`Entering recovery via button press <ug_nrf54h20_suit_recovery_enter_via_button>`, which demonstrates similar request-based entry.

The implementation of this feature is available in the :file:`nrf/subsys/suit/app_tools/recovery_button` directory.  
This directory contains a reference design that can be adapted as needed.

To request the device to enter recovery mode, invoke the ``suit_foreground_dfu_required()`` function.  
```

Reply:
Additional to these changes, line 252:

"For additional details, see :ref:`Entering recovery mode through button press <ug_nrf54h20_suit_recovery_enter_through_button>`, which demonstrates similar request-based entry."

Reply:
The suggested change has a bit different meaning than I intended - the recovery_button is a module which uses the described feature (entering recovery through a request - by the way, it uses it exactly, it does not use a "similar entry") - however it is not strictly a reference design/sample, it is a full-fledged feature. However, it is currently the only code which uses this feature, so I wanted to suggest to the documentation reader that if he wants to see an example, this module is the best place to start with.

I've modified the text a bit, please take a look now